core/src/main/java/io/questdb/client/impl/ConfStringParser.java

```diff
@@ -0,0 +1,172 @@
+     */
+    public static boolean hasNext(CharSequence input, int pos) {
+        if (pos == -1) {
+            return false;
+        }
+        return pos != input.length();
+    }
+
+    /**
+     * Parses key from input and writes it to output.
+     * <p>
+     * Call this method only after a previous call to {@link #hasNext(CharSequence, int)} returned
+     * true.
+     * <p>
+     * This method will return a negative value if it encounters an error. In this case, the error
+     * message will be written to the output sink.
+     *
+     * @param input  input
+     * @param pos    position handler
+     * @param output output
+     * @return next position handler, or negative value if error
+     */
+    public static int nextKey(CharSequence input, int pos, StringSink output) {
+        output.clear();
+        if (pos == -1) {
+            return pos;
+        }
+        int n = input.length();
+        int start = pos;
+        for (; pos < n; pos++) {
+            char c = input.charAt(pos);
+            if (c == '=') {
+                if (pos == start) {
+                    output.put("empty key");
+                    return -1;
+                }
+                output.put(input, start, pos);
+                return pos + 1;
+            } else if (c == ';') {
+                output.put("incomplete key-value pair before end of input at position ").put(pos);
+                return -1;
+            } else if (invalidIdentifierChar(c)) {
+                output.put("key must be consist of alpha-numerical ascii characters and underscore, not '").put(c).put("' at position ").put(pos);
+                return -1;
+            }
+
+        }
+        output.clear();
+        output.put("incomplete key-value pair before end of input at position ").put(pos);
+        return -1;
+    }
+
+    /**
+     * Parses schema name from input. Schema name must start with schema type, e.g. http::.
+     * <p>
+     * This is the starting point for all configuration parsing. It returns a position handler
+     * that can be used to parse the rest of the configuration.
+     * <p>
+     * When parsing configuration, the parser will return -1 if it encounters an error. In this
+     * case, the error message will be written to the output sink.
+     *
+     * @param input  input
+     * @param output output
+     * @return position handler, or negative value if error
+     */
+    public static int of(CharSequence input, StringSink output) {
+        output.clear();
+        char lastChar = 0;
+        for (int i = 0, n = input.length(); i < n; i++) {
+            char c = input.charAt(i);
+            if (lastChar == ':') {
+                if (c == ':') {
+                    if (i == 1) {
+                        output.put("schema is empty");
+                        return -1;
+                    }
+                    if (i == n - 1) {
+                        output.put("missing trailing semicolon at position ").put(i);
```

Comment:
Confusing error message.

I suspect this should say something like "missing parameters"

Reply:
see this test: https://github.com/questdb/questdb/blob/d6d49a86e84e7af2f6bbd46bf400e65b9f63f999/core/src/test/java/io/questdb/test/client/impl/ConfStringParserTest.java#L130-L138

Reply:
fixed